xbmc/utils/LangCodeExpander.h

```diff
@@ -140,6 +141,20 @@ class CLangCodeExpander
   std::vector<std::string> GetLanguageNames(LANGFORMATS format = ISO_639_1,
                                             LANG_LIST list = LANG_LIST::DEFAULT);
 
+  /** \brief Performs various ISO 3166-1 lookups.
+  *          Input can be alpha-2, alpha-3 or name.
+  *          If the input is alpha-2 or alpha-3 then the
+  *          name will be returned unless returnCode is true,
+  *          then the opposite code will be returned.
+  *          If the input is a name, then alpha-3 is returned
+  *          unless returnCode is true, then alpha-2 is returned.
+  *   \param[in] saught The code or name being saught.
+  *   \param[out] returned The code or name being returned.
+  *   \param[in] returnCode Control the value returned.
+  *   \return true if the conversion succeeded, false otherwise.
+  */
+  bool LookupISO31661(std::string_view saught, std::string& returned, bool returnCode = false);
```

Comment:
This just seems bound to cause problems. Maybe look to use std::optional<std::string> as the return type, and drop the returnCode shenanigans (and returned)

Reply:
@fuzzard - Thanks for having a look at this.

This PR was submitted 3 months ago (no offence intended), so I'm a little vague on some of the details.  I will have to reacquaint myself with the code and get back to you.

Reply:
Yeah, unfortunately there are large chunks of the codebase that are, for all intents and purposes, unmaintained. When someone PR's to such areas it often takes considerable time and effort by active team members to go and understand these sections to be even able to evaluate a PR. Unfortunately they often stand for extended periods with little feedback.

Reply:
> This just seems bound to cause problems. Maybe look to use std::optional<std::string> as the return type, and drop the returnCode shenanigans (and returned)

@fuzzard - I reviewed the code and re-read the PR description, etc, and I'm not completely sure what 'shenanigans' you are referring to.  Can you please provide some more details?

With your 'std::optional<std::string>' comment:  Are you suggesting that the method should be defined as follows:

`std::optional<std::string> LookupISO31661(std::string_view sought, bool returnCode = false);`

… and that it should be called (for example) as follows:

```
alpha3_code = LangCodeExpander.LookupISO31661(alpha2_code, true);
language_name = LangCodeExpander.LookupISO31661(alpha2_code, false);
```

If so, this would appear to be inconsistent with the existing 'lookup' (and a great number of other) methods in this class.  There are a few that return a string, but most return a bool with the returned value being passed back via a pointer in the argument list.

I promise that I'm not trying to be argumentative or obtuse, I'm just trying to reconcile my interpretation of your comment with the style of the new method and similar methods already in the class.